Client/game_sa/CAEVehicleAudioEntitySA.cpp

```diff
@@ -53,15 +55,80 @@ void CAEVehicleAudioEntitySA::TurnOnRadioForVehicle()
 
 void CAEVehicleAudioEntitySA::StopVehicleEngineSound(unsigned char ucSlot)
 {
-    DWORD          dwFunc = FUNC_CAESound__Stop;
     tVehicleSound* pVehicleSound = &m_pInterface->m_aEngineSounds[ucSlot];
     if (pVehicleSound->m_pSound)
+        ((void(__cdecl*)(CAESound*))FUNC_CAESound__Stop)(pVehicleSound->m_pSound);
+}
+
+int CAEVehicleAudioEntitySA::StoppedUsingBankSlot(int bankId)
+{
+    return ((int(__cdecl*)(int))FUNC_CAEVehicleAudioEntity__StoppedUsingBankSlot)(bankId);
+}
+
+short CAEVehicleAudioEntitySA::RequestBankSlot(unsigned short bankId)
+{
+    return ((short(__cdecl*)(short))FUNC_CAEVehicleAudioEntity__RequestBankSlot)(bankId);
+}
+
+short CAEVehicleAudioEntitySA::DemandBankSlot(unsigned short bankId)
+{
+    return ((short(__cdecl*)(short))FUNC_CAEVehicleAudioEntity__DemandBankSlot)(bankId);
+}
+
+void CAEVehicleAudioEntitySA::CancelVehicleEngineSound(short engineSoundStateID)
+{
+    return ((void(__thiscall*)(void*, short))FUNC_CAEVehicleAudioEntity__CancelVehicleEngineSound)((void*)m_pInterface, engineSoundStateID);
+}
+
+void CAEVehicleAudioEntitySA::MakeSureEngineSoundsAreLoaded()
+{
+    // TODO: Check if this is the right way to do this
+    //       (aka should we write a function to load the sound bank, and all that, or this hack is just fine)
```

Comment:
```suggestion
    // CAEVehicleAudioEntity::JustGotInVehicleAsDriver loads the engine sound banks
```
If it works and has no side effects, then it's totally fine.